src/link.tsx

```diff
@@ -18,7 +18,7 @@ export function Link(props: React.ComponentProps<typeof NextLink>) {
         navigate((as || href) as string, { scroll: scroll ?? true });
       }
     },
-    [props.onClick, href, as, replace, scroll]
+    [props.onClick, href, as, replace, scroll, router]
```

Comment:
you could use those more precise deps if you want, as it is only those that are used inside the function

```suggestion
    [props.onClick, href, as, replace, scroll, router.replace, router.push]
```